romancal/resample/resample.py

```diff
@@ -732,30 +734,40 @@ def gwcs_into_l3(model, wcsinfo):
     by indexing. This is fragile and will be a source of issues.
     """
     l3_wcsinfo = model.meta.wcsinfo
-    transform = wcsinfo.forward_transform
+    transform = wcs.forward_transform
 
     # Basic WCS info
     l3_wcsinfo.projection = "TAN"
+    l3_wcsinfo.pixel_shape = model.shape
     l3_wcsinfo.rotation_matrix = transform["pc_rotation_matrix"].matrix.value.tolist()
     l3_wcsinfo.dec_ref = transform.lat_6.value
     l3_wcsinfo.ra_ref = transform.lon_6.value
-    # l3_wcsinfo.x_ref = center of mosaic?
-    # l3_wcsinfo.y_ref = center of mosaic?
-    # l3_wcsinfo.s_region = from gwcs.bounding_box
-
-    # Mosaic info
-    # l3_wcsinfo.dec_center = from center of mosaic
-    # l3_wcsinfo.pixel_scale = ???
-    # l3_wcsinfo.pixel_scale_local = ???
-    # l3_wcsinfo.pixel_shape = image shape
-    # l3_wcsinfo.ra_center = from center of mosaic
-    # l3_wcsinfo.ra_corn1 = from bounding box
-    # l3_wcsinfo.ra_corn2 = from bounding box
-    # l3_wcsinfo.ra_corn3 = from bounding box
-    # l3_wcsinfo.ra_corn4 = from bounding box
-    # l3_wcsinfo.dec_corn1 = from bounding box
-    # l3_wcsinfo.dec_corn2 = from bounding box
-    # l3_wcsinfo.dec_corn3 = from bounding box
-    # l3_wcsinfo.dec_corn4 = from bounding box
-    # l3_wcsinfo.orientat = ???
-    # l3_wcsinfo.orientat_local = ???
+    l3_wcsinfo.x_ref = -transform.offset_0.value  # cdelt1
+    l3_wcsinfo.y_ref = -transform.offset_1.value  # cdelt2
+    l3_wcsinfo.pixel_scale = (transform.factor_3.value + transform.factor_4.value) / 2.0
+
+    world_center = wcs(*[v / 2.0 for v in model.shape])
+    world_center_plus = wcs(*[(v / 2.0) + 1 for v in model.shape])
+    world_delta = [w_plus - w for w_plus, w in zip(world_center_plus, world_center)]
+    l3_wcsinfo.ra_center = world_center[0]
+    l3_wcsinfo.dec_center = world_center[1]
+    l3_wcsinfo.pixel_scale_local = (world_delta[0] + world_delta[1]) / 2.0
```

Comment:
world_center_plus = wcs(center + 1, center + 1), so it corresponds to sqrt(2) pixels difference.  I start getting confused about what this is doing if things aren't aligned neatly with ra/dec.  These are also on the sphere, but we want pixel_scale to be in physical coordinates (i.e., if we're near dec 90, we still want pixels that are 1" away from one another to have values of ~1 here).  I think if we want to do this "right" we should do the following:
- get central pixel coordinates
- get one pixel up coordinates
- get one pixel to the right coordinates
- convert to SkyCoord
- get separations from central pixel to neighbors via c_center.separation(c_next)
- get angle between coordinates: c_center.position_angle(c_1) - c_center.position_angle(c_2).  Should be very close to 90!
- compute pixel area as separation_1 * separation_2 * np.abs(np.sin(angle))
- compute scale as sqrt(area)

That at least gives something that is rotation invariant and shouldn't have weird behavior near the poles, though I'm open to other suggestions.

Reply:
duh, completely left out the conversion to sky...

Reply:
We already have a method implementing the above. 
https://github.com/spacetelescope/stcal/blob/main/src/stcal/alignment/util.py#L402